yt_dlp/extractor/bilibili.py

```diff
@@ -1,509 +1,391 @@
+            'timestamp': reply.get('ctime'),
+            'parent': reply.get('parent') or 'root',
+        }
+        for children in map(self._get_all_children, reply.get('replies') or []):
+            yield from children
+
+    def extract_common_info(self, video_id, aid, cid, initial_state, play_info):
+        return {
+            'duration': float_or_none(play_info.get('timelength'), scale=1000),
+            'subtitles': self.extract_subtitles(video_id, initial_state, cid),
+            '__post_extractor': self.extract_comments(aid),
+        }
+
+
+class BiliBiliIE(BilibiliBaseIE):
+    _VALID_URL = r'''(?x)https?://www\.bilibili\.com/video/
                         (?:
-                            (?:
-                                video/[aA][vV]|
-                                anime/(?P<anime_id>\d+)/play\#
-                            )(?P<id>\d+)|
-                            (s/)?video/[bB][vV](?P<id_bv>[^/?#&]+)
+                            [aA][vV](?P<aid>[^/?#&]+)
+                           |[bB][vV](?P<bv_id>[^/?#&]+)
                         )
-                        (?:/?\?p=(?P<page>\d+))?
-                    '''
+                  '''
 
     _TESTS = [{
-        'url': 'http://www.bilibili.com/video/av1074402/',
-        'md5': '7ac275ec84a99a6552c5d229659a0fe1',
-        'info_dict': {
-            'id': '1074402_part1',
-            'ext': 'mp4',
-            'title': '【金坷垃】金泡沫',
-            'uploader_id': '156160',
-            'uploader': '菊子桑',
-            'upload_date': '20140420',
-            'description': 'md5:ce18c2a2d2193f0df2917d270f2e5923',
-            'timestamp': 1398012678,
-            'tags': ['顶上去报复社会', '该来的总会来的', '金克拉是检验歌曲的唯一标准', '坷垃教主', '金坷垃', '邓紫棋', '治愈系坷垃'],
-            'bv_id': 'BV11x411K7CN',
-            'cid': '1554319',
-            'thumbnail': 'http://i2.hdslb.com/bfs/archive/c79a8cf0347cd7a897c53a2f756e96aead128e8c.jpg',
-            'duration': 308.36,
-        },
-    }, {
-        # Tested in BiliBiliBangumiIE
-        'url': 'http://bangumi.bilibili.com/anime/1869/play#40062',
-        'only_matching': True,
-    }, {
-        # bilibili.tv
-        'url': 'http://www.bilibili.tv/video/av1074402/',
-        'only_matching': True,
-    }, {
-        'url': 'http://bangumi.bilibili.com/anime/5802/play#100643',
-        'md5': '3f721ad1e75030cc06faf73587cfec57',
-        'info_dict': {
-            'id': '100643_part1',
-            'ext': 'mp4',
-            'title': 'CHAOS;CHILD',
-            'description': '如果你是神明，并且能够让妄想成为现实。那你会进行怎么样的妄想？是淫靡的世界？独裁社会？毁灭性的制裁？还是……2015年，涩谷。从6年前发生的大灾害“涩谷地震”之后复兴了的这个街区里新设立的私立高中...',
-        },
-        'skip': 'Geo-restricted to China',
-    }, {
-        'url': 'http://www.bilibili.com/video/av8903802/',
+        'url': 'https://www.bilibili.com/video/BV13x41117TL',
         'info_dict': {
-            'id': '8903802_part1',
-            'ext': 'mp4',
+            'id': '13x41117TL',
```

Comment:
If you are changing the archive key, add `_old_archive_ids` to the infodict. Same for all URL patterns
```
❯ yt-dlp -O "%(extractor_key)s %(id)s" https://www.bilibili.com/video/BV13x41117TL
BiliBili 8903802_part1
```

Reply:
get

Reply:
I changed the test url, it didn't break compatibility

Reply:
> I changed the test url, it didn't break compatibility

oh, I'm wrong, I will add _old_archive_ids

Reply:
the old behavior is use the id in URL, no matter whether it is aid or bvid, and always add `_partXX`, 
so the test should be:
```
❯ yt-dlp -O "%(extractor_key)s %(id)s" https://www.bilibili.com/video/BV13x41117TL
BiliBili 13x41117TL_part1
```